File: lib/core/network/api_response.dart
class ApiResponse<T> {
  final bool success;
  final String? message;
  final T? data;
  final int? statusCode;
  final Map<String, dynamic>? meta;

  ApiResponse({
    required this.success,
    this.message,
    this.data,
    this.statusCode,
    this.meta,
  });

  factory ApiResponse.fromJson(
    Map<String, dynamic> json,
    T Function(dynamic) fromJsonT,
  ) {
    return ApiResponse<T>(
      success: json['success'] ?? true,
      message: json['message'] as String?,
      data: json['data'] != null ? fromJsonT(json['data']) : null,
      statusCode: json['status_code'] as int?,
      meta: json['meta'] as Map<String, dynamic>?,
    );
  }
}

File: backend/src/utils/apiResponse.ts
import { Response } from 'express';

export function ok(res: Response, data: unknown, status = 200, message?: string) {
  return res.status(status).json({ success: true, message: message ?? null, data, status_code: status, meta: null });
}

export function fail(res: Response, message: string, status = 400) {
  return res.status(status).json({ success: false, message, data: null, status_code: status, meta: null });
}
